apps/console/src/features/identity-providers/components/wizards/microsoft/microsoft-authentication-provider-create-wizard.tsx

```diff
@@ -0,0 +1,535 @@
+
+        const { t } = useTranslation();
+        const { getLink } = useDocumentation();
+
+        const [ alert, setAlert, alertComponent ] = useWizardAlert();
+
+        const config: ConfigReducerStateInterface = useSelector((state: AppState) => state.config);
+
+        const [ currentWizardStep, setCurrentWizardStep ] = useState<number>(currentStep);
+        const [ wizStep, setWizStep ] = useState<number>(0);
+        const [ totalStep, setTotalStep ] = useState<number>(0);
+        const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+        const [ openLimitReachedModal, setOpenLimitReachedModal ] = useState<boolean>(false);
+
+        const eventPublisher: EventPublisher = EventPublisher.getInstance();
+
+        /**
+        * Track wizard steps from wizard component.
+        */
+        useEffect(() => {
+            setCurrentWizardStep(wizStep + 1);
+        }, [ wizStep ]);
+
+        /**
+        * Creates a new identity provider.
+        *
+        * @param identityProvider - Identity provider object.
+        */
+        const createNewIdentityProvider = (identityProvider: IdentityProviderInterface): void => {
+
+            setIsSubmitting(true);
+
+            createIdentityProvider(identityProvider)
+                .then((response) => {
+                    eventPublisher.publish("connections-finish-adding-connection", {
+                        type: componentId
+                    });
+
+                    dispatch(addAlert({
+                        description: t("console:develop.features.authenticationProvider.notifications.addIDP." +
+                        "success.description"),
+                        level: AlertLevels.SUCCESS,
+                        message: t("console:develop.features.authenticationProvider.notifications.addIDP." +
+                        "success.message")
+                    }));
+
+                    // The created resource's id is sent as a location header.
+                    // If that's available, navigate to the edit page.
+                    if (!isEmpty(response.headers.location)) {
+                        const location = response.headers.location;
+                        const createdIdpID = location.substring(location.lastIndexOf("/") + 1);
+
+                        onIDPCreate(createdIdpID);
+
+                        return;
+                    }
+
+                    // Since the location header is not present, trigger callback without the id.
+                    onIDPCreate();
+                })
+                .catch((error) => {
+
+                    const identityAppsError: IdentityAppsError = identityProviderConfig.useNewConnectionsView
+                        ? IdentityProviderManagementConstants.ERROR_CREATE_LIMIT_REACHED
+                        : IdentityProviderManagementConstants.ERROR_CREATE_LIMIT_REACHED_IDP;
+
+                    if (error.response.status === 403 &&
+                    error?.response?.data?.code ===
```

Comment:
Check indentation. Might fit 2 out of 3 lines in one line

Reply:
This cannot be done as the line length exceeds 120